fontir/src/stateset.rs

```diff
@@ -13,32 +12,28 @@ use std::{
 
 /// Helps to identify changes in a set of stateful things.
 #[derive(Serialize, Deserialize, Debug, Default, Clone, PartialEq)]
-#[serde(from = "StateSetSerdeRepr", into = "StateSetSerdeRepr")]
 pub struct StateSet {
-    pub(crate) entries: HashMap<StateIdentifier, State>,
-}
-
-// The key for a stateful thing of some specific type
-#[derive(Debug, Clone, Hash, PartialEq, Eq)]
-pub enum StateIdentifier {
-    File(PathBuf),
-    Memory(String),
+    // our entries can either be files on disk or items in memory, but
+    // as a simplification just use PathBuf as the key type; we know
+    // if it's an actual file or not based on the value type.
+    pub(crate) entries: HashMap<PathBuf, State>,
 }
```

Comment:
Giving up the clear indication of state type seems unfortunate to me, I'd rather we not do that and if it means a bit more serde conversion that's ok

Reply:
Yea, I tried a bunch of other things first, but this is the simplest, and its worth remembering that the type is preserved in the `State` enum.